### PR TITLE
[IBCDPE-576] Adds LENS file staging Metaflow DAG

### PR DESCRIPTION
**Pull request by BWMac**
This PR adds the first local development script to the `local` directory.

The script will eventually be used to orchestrate an entire `LENS` workflow run on Nextflow Tower. For now, I have added the first step to enable reproducible file staging.

The dataset and input files can be found [here](https://www.synapse.org/#!Synapse:syn51748103).

This script ran successfully and the staged files are [here](https://s3.console.aws.amazon.com/s3/buckets/iatlas-project-tower-bucket?region=us-east-1&prefix=LENS/synstage/&showversions=false).

### CODE REVIEW

File: local/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+### Environment Variables
+
+In order for the scripts leveraging `py-orca` to connect to Nextflow Tower, you will need to configure the following environment variables:
+    - `NEXTFLOWTOWER_CONNECTION_URI`
+    - `SYNAPSE_CONNECTION_URI`
+    - `AWS_ACCESS_KEY_ID`
+    - `AWS_SECRET_ACCESS_KEY`
+    - `AWS_SESSION_TOKEN`
+    * Optionally, you can use `aws sso login` in place of the AWS environment variable credentials if you have pre-configured AWS profiles already.
```
Comment (thomasyu888) [inline local/README.md]: In that case, you would use `AWS_PROFILE` correct?

File: local/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+You can copy `local/.env.example` into a local `local/.env` file, replace the placeholders with your credentials, and run `source local/.env` in your terminal.
+
+## LENS
+
+Steps to run LENS workflow:
+1. Create necessary input files and upoload them to Synapse.
+    - The metaflow DAG requires 2 pre-formatted input files
+        - an input CSV file containing the LENS manifest information along with the Synapse URI's needed for `nf-synstage`
+    - A dataset YAML file containing a dataset ID and the Synapsee ID for the above described input CSV file
```
Comment (thomasyu888) [inline local/README.md]:
```suggestion
        - an input CSV file containing the LENS manifest information along with the Synapse URI's needed for `nf-synstage`
        - A dataset YAML file containing a dataset ID and the Synapse ID for the above described input CSV file
```

Comment (thomasyu888) [reply local/README.md]: Nit: you might want to provide an example of a Synapse URI to show how it differs from just a Synapse ID.

File: local/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+2. Set up your environment.
+    - Use the `.env.example` file to create your own `.env` file with all of the necessary environment variables defined.
+    - Source your `.env` file 
+    ```
+    source .env
+    ```
+3. Run the `lens.py` script with appropriate inputs
+```
+python3 local/lens.py run --dataset_id <yaml-dataset-synapse-id> --s3_prefix s3://<your-s3-bucket>/<your-s3-subdirectory>
```
Comment (thomasyu888) [inline local/README.md:42]: Do you have the full command you ran here for this particular task?

**Review by thomasyu888 — APPROVED** [review]
💯 Its great to see being used - I must admit, its nice to be able to use something like metaflow for local execution.

Going to pre-approve, but just some comments!

File: local/README.md (hunk 3 above)
Comment (BWMac) [reply local/README.md:42]: Yes, it is at the bottom of `lens.py`

File: local/README.md (hunk 1 above)
Comment (BWMac) [reply local/README.md]: Right. I added `AWS_PROFILE` to the example `.env` file and added a description here.

**Pull request merged**